### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by HeyItsGwen**
Merge feature branch into development branch for code review

### CODE REVIEW

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      elsif pattern.is_a?(Class)
+        self.my_each { |x| return true if x.is_a?(pattern) }
+      else
+        self.my_each { return true if x == pattern }
+      end
+    else
+      self.my_each { |x| return true if x }
+    end
+    false
+  end
```
Comment (wathigo) [inline enumerables.rb]: **[REQUIRED]**
* When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern.
```
Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
     
     NameError:
       undefined local variable or method `x' for ["dog", "door", "rod", "blade"]:Array
```

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      elsif pattern.is_a?(Class)
+        self.my_each { |x| return false if x.is_a?(pattern) }
+      else
+        self.my_each { |x| return false if x == pat }
+      end
+    else
+      self.my_each { |x| return false if x }
+    end
+    true
+  end
```
Comment (wathigo) [inline enumerables.rb]: **[REQUIRED]**
* When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern.
```
Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
     
     NameError:
       undefined local variable or method `pat' for ["dog", "door", "rod", "blade"]:Array
       Did you mean?  pack
```
* Returns true if the block never returns true for all elements.
```
Failure/Error: expect(array.my_none?(&true_block)).to eq(array.none?(&true_block))
     
       expected: true
            got: false
     
       (compared using ==)
```

File: enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    return self.length unless block_given?
+
+    if elem.nil?
+      return self.my_select { |x| yield(x) }.length
+    else
+      count = 0
+      self.my_each { |x| count += 1 if x == elem }
+      return count
+    end
+  end
```
Comment (wathigo) [inline enumerables.rb]: **[REQUIRED]**
* Counts the number of items in enum that are equal to item if an argument is given.
```
Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)
     
       expected: 13
            got: 100
     
       (compared using ==)
```

**Review by wathigo — CHANGES_REQUESTED** [review]
# Changes Required :smile: 
## Hi Team :wave:  
This is a great start!
We only have four issues we need to address, see the comments :point_down: 
### After implementing the changes, submit another request for review.

**Happy Coding :computer: 
Simon Wathigo**

File: enumerables.rb [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        self.my_each { |x| return false if pattern.match(x.to_s) }
+      elsif pattern.is_a?(Class)
+        self.my_each { |x| return false if x.is_a?(pattern) }
+      else
+        self.my_each { |x| return false if x == pattern }
+      end
+    else
+      self.my_each { |x| return false if x }
+    end
+    false
```
Comment (ebukaume) [inline enumerables.rb]: Are you sure about this? Should you be returning `true`? :wink:

**Review by ebukaume — CHANGES_REQUESTED** [review]
## Changes Required! :stop_sign: 

Hi @HeyItsGwen :wave: 

Nice job attempting to fix the fix as requested by @wathigo. Unfortunately, the problem seems to be there.

- [ ] `my_none?` still returns a wrong answer when a proc is passed.
Also, `[nil, false].my_none?` returns `false` instead of `true`. 

We have same issue when a Class, Regex, and primitives are passed in too.

Are you sure you should be returning false on `line 85`?

**Kindly submit another code review form after making the corrections**

Happy Coding :v: 
**Ebuka Umeokonkwo**
 [Github](https://github.com/ebukaume) | [Linkedin](https://www.linkedin.com/in/ebukaume) | [Slack](https://app.slack.com/team/UFDKXE68L) | [Twitter](https://twitter.com/ebukaume)

**Review by ebukaume — CHANGES_REQUESTED** [review]
Hi @HeyItsGwen 

You still didn't address this issue :point_down: 

- [ ] `my_none?` still returns a wrong answer when a proc is passed.

**Make the fix, check that is working for you, then submit another code review**

Happy Coding :v: 
**Ebuka Umeokonkwo**
 [Github](https://github.com/ebukaume) | [Linkedin](https://www.linkedin.com/in/ebukaume) | [Slack](https://app.slack.com/team/UFDKXE68L) | [Twitter](https://twitter.com/ebukaume)

**Review by ebukaume — APPROVED** [review]
Congratulations @HeyItsGwen :champagne: 

**You can now merge this branch.**

Happy Coding :v: 
**Ebuka Umeokonkwo**
 [Github](https://github.com/ebukaume) | [Linkedin](https://www.linkedin.com/in/ebukaume) | [Slack](https://app.slack.com/team/UFDKXE68L) | [Twitter](https://twitter.com/ebukaume)

**Pull request merged**